src/main/java/uk/gov/hmcts/ccd/domain/model/common/CaseReferenceUtils.java

```diff
@@ -11,4 +13,20 @@ public static String formatCaseReference(String caseReference) {
         }
         return caseReference;
     }
+
+    public static boolean checkRegex(final String value) {
+        final String regularExpression = "(?:^[0-9]{16}$|^\\d{4}-\\d{4}-\\d{4}-\\d{4}$)";
+        return value.matches(regularExpression);
+    }
+
+    public static Optional<String> getFormatCaseReference(Optional<String> caseReference) {
+
+        if (caseReference.isPresent()) {
+            if (!CaseReferenceUtils.checkRegex(caseReference.get())) {
+                return Optional.of("0000000000000000");
```

Comment:
is this safe?

Reply:
I also wonder whether there's a better way to do this, so someone searches for a non valid reference, we don't execute the query and immediately return empty results. Your solution executes the query.

Reply:
I think it would be better to add case reference validation here:
https://github.com/hmcts/ccd-data-store-api/blob/80adc76067063ba3c3600fb3e0674b41bfe5426f/src/main/java/uk/gov/hmcts/ccd/domain/service/search/DefaultSearchOperation.java#L26

so that if the user is searching by case reference and is an invalid one, we just return empty results without executing the query on the DB

Reply:
That one it is important !